annotations.py

```diff
@@ -189,12 +194,28 @@ def to_HTML(self, header=''):
                     dt_fgcolor = COLOR_MAP[agroup.highlightcolor]['fg']
                 except:
                     if agroup.highlightcolor is None:
-                        msg = "No highlight color specified, using Default"
+                        msg = "No highlight color specified, using default"
+                        dt_bgcolor = plugin_prefs.get('appearance_highlight_bg')
+                        dt_fgcolor = plugin_prefs.get('appearance_highlight_fg')
+                    elif html_color_pattern.match(agroup.highlightcolor):
+                        msg = "Found valid HTML color '%s', using" % agroup.highlightcolor
+                        dt_bgcolor = agroup.highlightcolor
+
+                        # Currently, annotations only store the foreground color. If it's not a known color, then
+                        # we should pick the background color as black or white based on the overall lightness of
+                        # the color.
+                        color = QColor(dt_bgcolor)
+                        if color.lightness() > 128:
+                            dt_fgcolor = "#000000"
+                        else:
+                            dt_fgcolor = "#FFFFFF"
```

Comment:
Example of how this works:

![image](https://user-images.githubusercontent.com/429763/163741644-8e64e769-7ecd-4f65-a6e9-aadac6066ce9.png)

As you can see, white gets a black background. The others get white.